src/mediarithmics/plugins/computed-field/ComputedFieldBasePlugin.ts

```diff
@@ -7,52 +7,97 @@ export type OnUpdatePluginStatus = 'ok' | 'error';
 export interface PluginResponse {
   status: OnUpdatePluginStatus;
 }
-export interface OnUpdatePluginResponse<D> extends PluginResponse {
-  data: D;
+export interface OnUpdatePluginResponse<S> extends PluginResponse {
+  data: {
+    state: S;
+  };
+}
+
+export interface OnUpdateBatchPluginResponse<S> extends PluginResponse {
+  data: {
+    state: S;
+  };
+}
+
+export interface BuildResultPluginResponse<R> extends PluginResponse {
+  data: {
+    result: R;
+  };
 }
 
-export interface OnUpdateBatchPluginResponse<D> extends PluginResponse {
-  data: D[];
+export enum DataType {
+  USER_ACTIVITY,
+  USER_PROFILE,
+  COMPUTED_FIELD,
 }
 
-export interface BuildResultPluginResponse<D> extends PluginResponse {
-  data: D;
+export enum Operation {
+  INSERT,
+  DELETE,
+  UPDATE,
 }
 
-export interface RequestData<S, D> {
+export interface BaseUserActivity {}
+export interface BaseUserProfile {}
+export interface BaseComputedField {}
+
+export interface Update {
+  dataType: DataType;
+  operation: Operation;
+  data: BaseUserActivity | BaseUserProfile | BaseComputedField;
+}
+
+export interface RequestData<S> {
   state: S;
-  data: D;
+  update: Update;
 }
 
-export interface RequestDataBatch<S, D> {
+export interface RequestDataBatch<S> {
   state: S;
-  data: D[];
+  updates: Update[];
 }
 
 export interface RequestResult<S> {
   state: S;
 }
 
-export abstract class ComputedFieldPlugin<State, Data, Result> extends BasePlugin {
+export abstract class ComputedFieldPlugin<
+  State,
+  Result,
+  UserActivity extends BaseUserActivity,
+  UserProfile extends BaseUserProfile,
+  ComputedField extends BaseComputedField,
+> extends BasePlugin {
   constructor() {
     super();
     this.initUpdateRoute();
     this.initUpdateBatchRoute();
     this.initBuildResultRoute();
   }
 
-  abstract onUpdate(state: State | null, data: Data): State | null;
-
-  abstract buildResult(state: State | null): {
-    state: State | null;
-    result: Result;
-  };
+  abstract onUpdateActivity(state: State | null, userActivity: UserActivity): State | null;
+  abstract onUpdateUserProfile(state: State | null, userProfile: UserProfile, operation: Operation): State | null;
+  abstract onUpdateComputedField(state: State | null, computedField: ComputedField): State | null;
+
+  abstract buildResult(state: State | null): Result | null;
```

Comment:
Are we sure we want to remove state from this interface ?

Reply:
Yes